outputs/gcp.go

```diff
@@ -32,17 +32,27 @@ func NewGCPClient(config *types.Configuration, stats *types.Statistics, promStat
 	var storageClient *storage.Client
 
 	if config.GCP.PubSub.ProjectID != "" && config.GCP.PubSub.Topic != "" {
-		credentials, err := google.CredentialsFromJSON(context.Background(), []byte(googleCredentialsData), pubsub.ScopePubSub)
-		if err != nil {
-			log.Printf("[ERROR] : GCP PubSub - %v\n", "Error while loading GCP Credentials")
-			return nil, errors.New("Error while loading GCP Credentials")
-		}
-		pubSubClient, err := pubsub.NewClient(context.Background(), config.GCP.PubSub.ProjectID, option.WithCredentials(credentials))
-		if err != nil {
-			log.Printf("[ERROR] : GCP PubSub - %v\n", "Error while creating GCP PubSub Client")
-			return nil, errors.New("Error while creating GCP PubSub Client")
+		if googleCredentialsData != "" {
+			credentials, err := google.CredentialsFromJSON(context.Background(), []byte(googleCredentialsData), pubsub.ScopePubSub)
+			if err != nil {
+				log.Printf("[ERROR] : GCP PubSub - %v\n", "Error while loading GCP Credentials")
```

Comment:
note: we need to log the actual error as well, will do a PR to update all those errors.
not needed to do here.

Reply:
Anything I can do in there to make your life easier?